var/spack/repos/builtin/packages/fftw/package.py

```diff
@@ -149,6 +150,12 @@ def configure(self, spec, prefix):
             options.append('--enable-mpi')
 
         # SIMD support
+        if 'default' in self.spec.variants['simd'].value:
+            if 'x86_64' in spec.architecture.target.lower():
+                self.spec.variants['simd'].value = 'sse2,avx,avx2'
```

Comment:
@tkameyama this will likely cause problems: generally we cannot set variants outside of concretization, since that can lead to changing Spec hashes. Currently, there isn't support for converting a list of values to configure options so you would have to replicate the `Autotools` logic here.

It can also be problematic to have a `default` variant value that has different meanings based on other Spec properties. It will not cause a build error though.

Would it be problematic to set this in a config file? If you want to help other users avoid the wrong `simd` settings, you could also add a check in this method to report an error if the architecture is not `x86_64` and `sse2` is used. Ideally you could encode this with `conflict` statements, but they cannot express a constraint like

`conflicts('simd=sse2', when='! target=x86_64') # when syntax does not currently support negation i.e. "!"`

Reply:
I change simd varialt default to generic-simd128,generic-simd256 (and rebase).

We can write not x86_64 and sse2 to flag_handler.
But cray platform is fails because the target from environment variable CRAY_CPU_TARGET.

Reply:
Sorry, tkameyama:fftw_on_aarch64 branch was broken.
I'm fixed the branch. And add check architecture and simdby flag_handler.